### ISSUE TITLE
Translate documentation into Arabic

### ISSUE THREAD

**Issue by BaseMax** [SalamLang/Salam#13]
We will ask someone who is a native Arabic

**Comment by yOOs-dz** [comment]
Hello @BaseMax,
I am a native Arabic speaker. Interested in participating to the project. How could I do for it ?
Please visit my new [repository ](https://github.com/yOOs-dz/R_tips_ar).

**Comment by jbampton** [comment]
Hello @yOOs-dz hope you are doing well.

We are back online with a fully redesigned system and and online playground.

Are you still interested in helping with translation for Arabic ?

If you are still interested there is a real position for you on the core team as "Translator".

Check out the new project and core team here:

https://github.com/SalamLang

We have roles for issue and PR triage so that might be an option in future as well.

Thanks once again

**Comment by yOOs-dz** [comment]
Hello @jbampton ,

I'm doing well, hope you are too. Thanks for the invitation, glad to see the project back with a fresh look! Also, sorry for the delayed response.

I did a small check on the project. In fact, I already contributed to a commit #121, two years ago! 
Yes, I'm interested in the Arabic translator role. I'd be glad to join the core team in that capacity.

Before I dive in, I would like to ask some questions about how to contribute to translation within the Salam language project, which elements I could work on, and everything related to the project's working methodology:
1. Is there a reference glossary for various terms and commands, for example, or should I create one?
2. How can translations be added for the contribution file, the README file, and others? Is there a methodology already followed, or will I help establish it? I see that the system is live in Persian, but I haven't found such glossary or methodological frame in the repository. That's in order to coordinate the work to be done in Arabic?
3. Is there also an issue tracker or board to follow the status of Arabic translation or should I help to create one?

On time commitment: I'm currently deep in my doctoral thesis, so my contribution will be steady but limited weekly. I would like to know what possibilities are available within the project team.


Looking forward to you to get started! Thank you again for your invitation.

**Comment by BaseMax** [comment]
سلام و علیکم
کیف حالک
شو اخبارک؟
انا علی من ایران فی المندینه الکاشان.

I started to leraning Arabic recently but I am not really good in it.

Here is the place of main keybords, currently we are fully supporting en and fa, you can start working on ar (Arabic).
https://github.com/SalamLang/Salam/blob/main/compiler/src/i18n/i18n_gettext.c

After that, next step is to add `@ar ....` for all package and function or variables we have in the `std` lib. we already did all of it for `@en ...` and `@fa ...` and they are working.

And at the end, we will add arabic into editor so you can test it in the live editor too :)

Many thanks.
Happy to see you again back to team.

**Comment by yOOs-dz** [comment]
وعليكم السلام و رحمة الله،

بخير و أنت علي، كيف حالك؟

يوسف من بسكرة، الجزائر

I wish you good luck learning Arabic! By participating in this project, I will also take my first steps towards learning Farsi. I enjoy learning languages. Thank you, Ali, for the explanation that clarified many things for me. I will get back to you as soon as I can.

It's my pleasure.

**Comment by yOOs-dz** [comment]
Hello @BaseMax , @jbampton ,

I hope you are doing well.

I am taking the time to understand the structure of the Salam project. I made a fork of the repository and cloned it before starting to work on it. I also ran some small tests on the playground.  Ali told me that one of the first steps is to provide an Arabic version of this file: [https://github.com/SalamLang/Salam/blob/main/compiler/src/i18n/i18n_gettext.c](https://github.com/SalamLang/Salam/blob/main/compiler/src/i18n/i18n_gettext.c)

This includes a relevant translation methodology with clear references to provide an adequate explanation (I can start working on that). It also provides a clear view of the file's structure and how it communicates with the project environment.

I would like to know if a separate file is needed for Arabic, what its name should be, and how it will be mentioned elsewhere in the project. I need a brief explanation of the structure of this part of the project to understand how to create the Arabic file and where to put it. 

In summary: What do I need to do to start translating this file to Arabic? 

Please note that I am not a programmer by trade, though I am familiar with Python and HTML and am proficient in R.

Thank you and good luck.

**Comment by BaseMax** [comment]
Many thanks for your support. I added Arabic and its base into the language. and you can see we have Arabic language in the editor as well now: https://salamlang.github.io/Salam/

But arabic keywords might are not great as I used AI to translate en/fa into ar. So this is mostly ready for you to review the AR in the compiler.

**Comment by BaseMax** [comment]
We have some arabic words at `editor/app.js`:

```
    ar: {
      title: "ساحة سلام",
      mode_app: "برنامج",
      mode_layout: "تخطيط",
      run: "تشغيل",
      autorun: "تشغيل تلقائي",
      editor: "المحرر",
      output: "الإخراج",
      preview: "معاينة",
      ready: "جاهز",
      running: "قيد التشغيل…",
      done: "تم",
      error: "خطأ",
      loading: "تحميل المترجم…",
      custom: "مخصص",
      examples: "أمثلة",
      g_console: "برامج طرفية",
      g_web: "صفحات وب",
      badge_web: "وب",
      theme_auto: "تلقائي",
      theme_light: "فاتح",
      theme_dark: "داكن",
      view_run: "تشغيل",
      view_tokens: "الرموز",
      view_ast: "الشجرة",
      view_symbols: "الرموز البرمجية",
      view_html: "HTML",
      view_css: "CSS",
      view_js: "JS",
      view_c: "كود C",
      view_llvm: "LLVM IR",
      copy: "نسخ",
      copied: "تم النسخ",
```

After that main keywords are located in `compiler/src/i18n/i18n_gettext.c` file.

The rest are at `compiler/std/**/*.salam` files.
